### PR TITLE
adding uct put and get over cuda-ipc (cache design pending)

### PR DESCRIPTION
**Pull request by Akshay-Venkatesh**
Seems like some changes from master branch has gone into this PR. Sorry about that.

### CODE REVIEW

**Comment by Akshay-Venkatesh** [comment]
@bureddy

File: src/tools/perf/libperf.c [hunk 1]
```diff
@@ -553,12 +553,10 @@ static ucs_status_t uct_perf_test_setup_endpoints(ucx_perf_context_t *perf)
...
-    if (iface_attr.cap.flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE) {
-        status = uct_iface_get_address(perf->uct.iface, iface_addr);
-        if (status != UCS_OK) {
-            ucs_error("Failed to uct_iface_get_address: %s", ucs_status_string(status));
-            goto err_free;
-        }
+    status = uct_iface_get_address(perf->uct.iface, iface_addr);
+    if (status != UCS_OK) {
+        ucs_error("Failed to uct_iface_get_address: %s", ucs_status_string(status));
+        goto err_free;
```
Comment (bureddy) [inline src/tools/perf/libperf.c:559]: why do you need this change? why can't you add UCT_IFACE_FLAG_CONNECT_TO_IFACE flag to your iface?

File: src/ucp/core/ucp_worker.c [hunk 2]
```diff
@@ -292,14 +292,16 @@ void ucp_worker_signal_internal(ucp_worker_h worker)
...
+    ucp_worker_h            worker           = (ucp_worker_h)arg;
+    ucp_ep_h                ucp_ep           = NULL;
+    uct_ep_h                aux_ep           = NULL;
+    uct_tl_resource_desc_t* tl_rsc;
+    uint64_t                dest_uuid UCS_V_UNUSED;
+    ucp_ep_h                ucp_ep_iter;
+    khiter_t                ucp_ep_errh_iter;
+    ucp_err_handler_cb_t    err_cb;
+    ucp_lane_index_t        lane, n_lanes, failed_lane;
+    ucp_rsc_index_t         rsc_index;
```
Comment (bureddy) [inline src/ucp/core/ucp_worker.c:304]: avoid indentation changes, so that it would be clear on actual changes

**Comment by Akshay-Venkatesh** [comment]
Hi, Devendar.

I'm not sure why these changes showed up because all the changes are under
src/uct/cuda/cuda_ipc

I'll revisit the PR and see if I've unintentionally added something.

**Pull request closed without merging**